xds/internal/xdsclient/xdsresource/unmarshal_cds.go

```diff
@@ -51,6 +51,12 @@ func UnmarshalCluster(opts *UnmarshalOptions) (map[string]ClusterUpdateErrTuple,
 }
 
 func unmarshalClusterResource(r *anypb.Any, f UpdateValidatorFunc, logger *grpclog.PrefixLogger) (string, ClusterUpdate, error) {
+	var err error
+	r, err = unwrapResource(r)
+	if err != nil {
+		return "", ClusterUpdate{}, fmt.Errorf("failed to unwrap resource: %v", err)
+	}
```

Comment:
Wouldn't the following work?

```golang
	r, err := unwrapResource(r)
	if err != nil {
		return "", ClusterUpdate{}, fmt.Errorf("failed to unwrap resource: %v", err)
	}
```

And same applies to all calls to `unwrapResource()`

Reply:
Yeah, that works!